### PR TITLE
Enhance timeout resolution logic to show more detailed information in dry-run mode

### PR DESCRIPTION
**Pull request by isseis**
This pull request focuses on enhancing the timeout resolution context for commands, especially in dry-run mode. The main goal is to provide more transparency in how timeout values are determined and displayed, making debugging and troubleshooting easier. The changes include unifying the timeout resolution logic, updating related tests, and improving dry-run output to show both the resolved timeout value and its source level.

### Timeout Resolution Logic Improvements

* Unified the timeout resolution logic by removing the old `ResolveEffectiveTimeout` function and updating all usages to the enhanced `ResolveTimeout`, which now returns both the resolved value and context information about the source level (e.g. "command", "global", "default"). [[1]](diffhunk://#diff-6f17b1b25bc43b576fd414ce6e0d4c4b0c02210200879ce041ca3cee2f4dfadeL96-L119) [[2]](diffhunk://#diff-e4c06ffb6c25587d9a1c86d7be4a925ee204d0a93a70044cc213f97a62ffa6feL20-R57)

* Updated the `TimeoutResolutionContext` to include metadata such as the level at which the timeout was set, the command name, and the group name, improving the ability to trace configuration sources.

### Dry-Run Output and Integration Tests

* Added new integration tests (`TestDryRun_TimeoutResolutionContext`) to verify that dry-run output correctly displays both the resolved timeout value and the level it was sourced from. This includes tests for command-level, global-level, and default timeouts in both human-readable and JSON formats. [[1]](diffhunk://#diff-92f390ecc41076addf027f3ebed4c58139b876d41338f013e7d56ebd46fbb948R1-R126) [[2]](diffhunk://#diff-de9f93d35f08589b701801a24f9d008ef29c2ae73f4ebd81b36906d4871a3771L295-R397)

* Updated the construction of `RuntimeCommand` to pass the group name, enabling richer context for timeout resolution and its reporting.

### Test Suite and Documentation Updates

* Refactored and unified timeout resolution test cases to use the new context-aware `ResolveTimeout`, ensuring all edge cases and source levels are covered. Removed tests for the deprecated `ResolveEffectiveTimeout`. [[1]](diffhunk://#diff-c6608e8e7ea5d83772086f39b331c8ca2b08fd729d5df58263f5773a1d28b287L10-R127) [[2]](diffhunk://#diff-51688a1d118f687cd0229bba021d8090b95ac3cd8cfa2b84712f4ee00021df19L278-L339)

* Added a detailed requirements document outlining the motivation, design, and expected effects of the timeout resolution context enhancement, including future extensibility for group-level timeouts.

### CODE REVIEW

File: cmd/runner/integration_dryrun_json_test.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+			cmd.Dir = "."
+
+			output, err := cmd.Output() // Use Output() instead of CombinedOutput() to get only stdout
+			require.NoError(t, err, "dry-run should succeed")
+
+			// Find the start of JSON output (skip debug information)
+			outputStr := string(output)
+			jsonStart := strings.Index(outputStr, "{")
+			require.NotEqual(t, -1, jsonStart, "should find JSON object start in output")
+			jsonOutput := strings.TrimSpace(outputStr[jsonStart:]) // Parse JSON output
```
[the remark this answers is not in the document]
Comment (isseis) [reply cmd/runner/integration_dryrun_json_test.go]: implemented more robust logic

File: internal/runner/resource/normal_manager_test.go [hunk 2]
```diff
@@ -197,10 +197,16 @@ func createTestCommandGroup() *runnertypes.GroupSpec {
...
 	commandTimeout := common.NewFromIntPtr(spec.Timeout)
 	globalTimeout := common.NewUnsetTimeout() // Tests typically don't need global timeout
-	effectiveTimeout := common.ResolveEffectiveTimeout(commandTimeout, globalTimeout)
+	effectiveTimeout, _ := common.ResolveTimeout(
+		commandTimeout,
+		common.NewUnsetTimeout(), // No group timeout in tests
+		globalTimeout,
+		spec.Name,
+		"test-group",
+	)
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/resource/normal_manager_test.go]: done
